src/toobj.c

```diff
@@ -949,7 +949,17 @@ void toObjFile(Dsymbol *ds, bool multiobj)
                 return;
 
             Symbol *s = toSymbol(vd);
-            unsigned sz = vd->type->size();
+            d_uns64 sz64 = vd->type->size(vd->loc);
+            if (sz64 == SIZE_INVALID)
+            {
+                vd->error("size overflow");
+                return;
+            }
+            if (sz64 >= 0x1000000)
```

Comment:
Please refactor the magic number into a proper constant and explain in the message below why the size is limited to `0x1000000` (i.e. why is it a problem).